src/govuk/components/error-summary/error-summary.js

```diff
@@ -109,7 +111,32 @@ ErrorSummary.prototype.getAssociatedLegendOrLabel = function ($input) {
     var legends = $fieldset.getElementsByTagName('legend')
 
     if (legends.length) {
-      return legends[0]
+      var $candidateLegend = legends[0]
```

Comment:
In theory we could write tests for this logic, where we do using puppeteer on the demo pages you've been testing, this could be useful in the future if we (hopefully not) encounter other edge cases.

Reply:
I've added to the existing tests to try and cover the two new examples that have been added to the review app.

Reply:
Epic! Thank you